### ISSUE TITLE
Refactor our concept names to be CamelCase

### ISSUE THREAD

**Issue by h-2** [seqan/seqan3#520]
Since it touches a lot files, this shouldn't be attempted before #498 is merged.

Our concept names are currently `seqan3::foo_bar_concept`, but to be more in line with the standard library, they should be called `seqan3::FooBar` in the future. Someone needs to make lots of replacements!

While we are at it, I would like to change some names, beyond just case:
```
seqan3::nucleotide_concept -> seqan3::NucleotideAlphabet
seqan3::aminoacid_concept -> seqan3::AminoacidAlphabet
... (the alphabets should end in "...Alphabet")

seqan3::const_iterable_concept -> seqan3::ConstIterableRange
```

Either of @joshuak94 or @MitraDarja could have a look at this!

**Comment by smehringer** [comment]
I like the name changes for alphabets :+1: (although I liked the `concept`-naming more to begin with :laughing:) 

> `seqan3::const_iterable_concept -> seqan3::ConstIterableRange`

This fits nicely with the Alphabet changes but it might be inconsistent with concept names where the object type (range in your case for const_iterable) is not known. e.g. what to do with `floating_point_concept -> FloatingPoint` vs `floating_point_concept -> FloatingPointValue`?

**Comment by rrahn** [comment]
@smehringer @h-2 @MitraDarja what is the state here? Can you make a list of modules already done/need refactorisation? This needs to be finished before the release.

**Comment by MitraDarja** [comment]
@rrahn I will work on this again, when I am done with fixing doxygen warnings due to missing default comments. 

Done:
aminoacid_concept -> AminoacidAlphabet
nucleotide_concept -> NucleotideAlphabet
quality_concept -> QualityAlphabet
semi_constexpr_alphabet_concept -> SemiConstexprAlphabet
constexpr_semi_alphabet_concept -> ConstexprSemialphabet
constexpr_alphabet_concept -> ConstexprAlphabet
semi_alphabet_concept -> Semialphabet
alphabet_concept -> Alphabet
rna_structure_concept -> RnaStructureAlphabet
char_adaption_concept -> CharAdaptationAlphabet
uint_adaption_concept -> UintAdaptionAlphabet
config_element_concept -> ConfigElement
cereal_archive_concept -> CerealArchive
cereal_output_archive_concept-> CerealOutputArchive
cereal_input_archive_concept -> CerealInputArchive
cerealisable_concept -> Cerealisable
weakly_equality_comparable_by_members_with_concept -> WeaklyEqualityComparableByMembersWith
weakly_ordered_by_members_with_concept -> WeaklyOrderedByMembersWith
implicitly_convertible_to_concept -> ImplicitlyConvertibleTo
explicitly_convertible_to_concept -> ExplicitlyConvertibleTo
convertible_to_concept -> ConvertibleTo
arithmetic_concept -> Arithmetic
floating_point_concept -> FloatingPoint
trivially_destructible_concept -> TriviallyDestructible
trivially_copyable_concept -> TriviallyCopyable
standard_layout_concept -> StandardLayout
trivial_concept -> Trivial
weakly_assignable_concept -> WeaklyAssignable
weakly_ordered_with_concept -> WeaklyOrderedWith
convertible_to_by_member_concept -> ConvertibleToByMember
sequence_file_output_format_concept -> SequenceFileOutputFormat
alignment_file_output_format_concept -> AlignmentFileOutputFormat
structure_file_output_format_concept -> StructureFileOutputFormat
output_format_concept-> OutputFormat
sequence_file_InputFormat -> SequenceFileInputFormat
structure_file_InputFormat -> StructureFileInputFormat
input_format_concept-> InputFormat
ostream_concept -> Ostream
istream_concept -> Istream
stream_concept -> Stream
type_list_of_structure_file_output_formats_concept -> TypeListOfStructureFileOutputFormats
type_list_of_structure_file_input_formats_concept -> TypeListOfStructureFileInputFormats
type_list_of_sequence_file_output_formats_concept ->TypeListOfSequenceFileOutputFormats
type_list_of_sequence_file_input_formats_concept ->TypeListOfSequenceFileInputFormats
type_list_of_alignment_file_output_formats_concept ->TypeListOfAlignmentFileOutputFormats
sequence_file_format_concept -> SequenceFileFormat
sequence_file_input_traits_concept -> SequenceFileInputTraits
sequence_file_format_out_concept -> SequenceFileFormatOut
parse_condition_concept -> ParseCondition

Missing are all concepts in alignment, range, search, std.
